packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -444,8 +596,14 @@ class _ModalBottomSheetState<T> extends State<_ModalBottomSheet<T>> {
           label: routeLabel,
           explicitChildNodes: true,
           child: ClipRect(
-            child: CustomSingleChildLayout(
-              delegate: _ModalBottomSheetLayout(animationValue, widget.isScrollControlled),
+            child: _BottomSheetLayoutWrapper(
+              onChildSizeChanged: (size) {
+                widget.route.sheetSizeNotifier.value = size;
```

Comment:
This should avoid accessing the notifier directly. better to call a function instead and keep sheetSizeNotifier private